domain/consensus/processes/blockprocessor/blocklogger/blocklogger.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-2017 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package blocklogger
+
+import (
+	"time"
+
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/util/mstime"
+)
+
+type blockType uint8
+
+const (
+	blockTypeHeader blockType = iota
+	blockTypeBody
+)
+
+var statsPerBlockType = map[blockType]*struct {
+	receivedLogBlocks int64
+	receivedLogTx     int64
+	lastBlockLogTime  time.Time
+}{
+	blockTypeHeader: {
+		lastBlockLogTime: time.Now(),
+	},
+	blockTypeBody: {
+		lastBlockLogTime: time.Now(),
+	},
+}
+
+// LogBlock logs a new block blue score as an information message
+// to show progress to the user. In order to prevent spam, it limits logging to
+// one message every 10 seconds with duration and totals included.
+func LogBlock(block *externalapi.DomainBlock) {
+	currentBlockType := blockTypeBody
+	if len(block.Transactions) == 0 {
+		currentBlockType = blockTypeHeader
```

Comment:
What if the previous 9.99s were nothing but headers and this is the first full block?
I propose that you print both blocks and headers in that case.